torch/nested/_internal/nested_tensor.py

```diff
@@ -178,23 +181,33 @@ def __tensor_flatten__(self):
         return inner_tensors, ctx
 
     @staticmethod
-    def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size, outer_stride):
+    def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size, outer_stride, original_tensors=None):
         # inner tensors: _values, _offsets, [_lengths]
         assert len(inner_tensors) >= 2 and len(inner_tensors) <= 3
         values = inner_tensors["_values"]
         offsets = inner_tensors["_offsets"]
         lengths = inner_tensors.get("_lengths", None)
         ragged_idx = meta["ragged_idx"]
-
-        # Note that we cannot simply check if is_fake(values) because
-        # during aot autograd, FunctionalTensors are not fake but hold
-        # symbolic sizes.
         ragged_source = offsets if lengths is None else lengths
-        if has_free_symbols(ragged_source) or has_free_symbols(values):
-            # Associate offsets or lengths (possibly fake, possibly functionalized)
-            # with the ragged_size.
-            ragged_size = outer_size[ragged_idx]
-            _tensor_symint_registry[ragged_source] = ragged_size
+        # for the functional to fake wrapping, don't need to do anything (since we rely on unwrapping)
+        # for the initial fakification? (different, and we've already saved it on the fake tensor)
+        # for the grad_output thing (same, and so we need to do this)
+        # for the runtime wrapper (different, and we just create a fresh one because we're in eager)
+        # the thing is... this is not enough?
+        if original_tensors:
+            orig_offsets = original_tensors.get("_offsets", None)
+            orig_lengths = original_tensors.get("_lengths", None)
+            orig_ragged_source = orig_offsets if orig_lengths is None else orig_lengths
+            if type(orig_ragged_source) is type(ragged_source):
+                from torch.nested._internal import union_find
+                assert orig_ragged_source is not None
+                uf = union_find.get_union_find()
+                uf.merge(ragged_source, orig_ragged_source)
+                # Two FakeTensor now share the same NestedInt?!
+                # Previously this was not the case. We also don't need to do
+                # the union find thing above?
+                ragged_source._nested_int_memo = orig_ragged_source._nested_int_memo
+                ragged_source._nested_int_memo_vc = ragged_source._version
```

Comment:
kinda unclear to me what's going on, and I thought this chunk of tensor_unflatten would become simpler after the revamp :/

Reply:
This logic is removed in the latest version!